common/src/main/java/org/apache/atlas/service/metrics/MetricUtils.java

```diff
@@ -0,0 +1,96 @@
+    private static final String STATUS = "status";
+    private static final String METHOD = "method";
+    private static final String SERVICE = "service";
+    private static final String INTEGRATION = "integration";
+    private static final String ATLAS_METASTORE = "atlas-metastore";
+    private static final String REGEX_URI_PLACEHOLDER = "\\[\\^/\\]\\+";
+    private static final String HTTP_SERVER_REQUESTS = "http.server.requests";
+    private static final String ATLAS_METRICS_URI_PATTERNS = "atlas.metrics.uri_patterns";
+
+    private static List<String> METRIC_URI_PATTERNS;
+    private static final PrometheusMeterRegistry METER_REGISTRY;
+
+    static {
+        METER_REGISTRY = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+        METER_REGISTRY.config().withHighCardinalityTagsDetector().commonTags(SERVICE, ATLAS_METASTORE, INTEGRATION, LOCAL);
+        Metrics.globalRegistry.add(METER_REGISTRY);
+    }
+
+    public MetricUtils() {
+        try {
+            METRIC_URI_PATTERNS = Arrays.asList(ApplicationProperties.get().getStringArray(ATLAS_METRICS_URI_PATTERNS));
+        } catch (Exception e) {
+            LOG.error("Failed to load 'atlas.metrics.uri_patterns from properties");
+        }
+    }
+
+    public Timer.Sample start(String uri) {
+        return matchCanonicalPattern(uri).isPresent() ? Timer.start(getMeterRegistry()) : null;
+    }
+
+    public void recordHttpTimer(Timer.Sample sample, String method, String rawPath, int code, String... additionalTags) {
+        if (Objects.isNull(sample)) {
+            return;
+        }
+        sample.stop(getTimer(HTTP_SERVER_REQUESTS, method, code, rawPath, additionalTags));
+    }
+
+    private Timer getTimer(String timerName, String method, int code, String rawPath, String... additionalTags) {
+        Tags tags = getTags(method, code, rawPath);
+        if (Objects.nonNull(additionalTags) && additionalTags.length > 0) {
+            tags = tags.and(additionalTags);
+        }
+        return Timer.builder(timerName)
+                .publishPercentiles(0.5,0.90,0.99)
+                .tags(tags)
+                .register(getMeterRegistry());
+    }
+
+    private Tags getTags(String httpMethod, int httpResponseStatus, String uri) {
+        return Tags.of(METHOD, httpMethod,
+                STATUS, String.valueOf(httpResponseStatus),
+                URI, matchCanonicalPattern(uri).get());
+    }
+
+    private Optional<String> matchCanonicalPattern(String uri) {
+        if (Objects.isNull(uri) || uri.isEmpty()) {
+            return Optional.empty();
+        }
+        if (uri.endsWith("/")) {
+            uri = uri.substring(0, uri.lastIndexOf("/"));
+        }
+        String updatedUrl = uri;
+        Optional<String> patternOp = METRIC_URI_PATTERNS.stream()
+                .filter(pattern -> updatedUrl.matches(pattern + "$"))
+                .findFirst();
+        return patternOp.map(s -> s.replaceAll(REGEX_URI_PLACEHOLDER, "*"));
```

Comment:
Optional: Check if we can prepare a map of [raw url string : replaced url strings] so that we ca avoid replace operations for each request

Reply:
see if we can do it with https://github.com/atlanhq/atlas-metastore/pull/2240#discussion_r1287986071